File: stackNqueue/stack.js
class Stack {
  constructor() {
    this._arr = []
  }
  push(item) {
    this._arr.push(item)
  }
  pop() {
    return this._arr.pop()
  }
  peek() {
    return this._arr[this._arr.length - 1]
  }
}

const stack = new Stack()

stack.push(1)
stack.push(2)
stack.push(3)
console.log(stack)
// Stack { _arr: [ 1, 2, 3 ] }

stack.pop()
console.log(stack)
// Stack { _arr: [ 1, 2 ] }

console.log(stack.peek())
// 2

File: stackNqueue/stack.py
from re import S
from turtle import st


class Stack():
    def __init__(self):
        self.arr = []

    def push(self, data):
        self.arr.append(data)

    def pop(self):
        self.arr.pop()


if __name__ == '__main__':
    stack = Stack()

    stack.push(4)
    stack.push(5)
    stack.push(2)
    stack.push(1)

    print(stack.arr)

    stack.pop()

    print(stack.arr)
